src/components/pages/devices/flyouts/deviceDetails/deviceDetails.js

```diff
@@ -0,0 +1,169 @@
+  }
+
+  applyRuleNames = (alarms, rules) =>
+    alarms.map(alarm => ({
+      ...alarm,
+      name: (rules[alarm.ruleId] || {}).name
+    }));
+
+  fetchAlarms = (deviceId) => {
+    this.setState({ isAlarmsPending: true });
+    TelemetryService.getAlarms({
+      limit: 5,
+      order: "desc",
+      devices: deviceId
+    })
+      .subscribe(
+      alarms => this.setState({ alarms, isAlarmsPending: false, alarmsError: undefined }),
+      alarmsError => this.setState({ alarmsError, isAlarmsPending: false })
+      );
+  }
+
+  render() {
+    const { t, onClose, device } = this.props;
+    const isPending = this.state.isAlarmsPending && this.props.isRulesPending;
+    const rulesGridProps = {
+      rowData: isPending ? undefined : this.applyRuleNames(this.state.alarms || [], this.props.rules || []),
+      t: this.props.t,
+      columnDefs: translateColumnDefs(this.props.t, this.columnDefs)
+    };
+    return (
+      <Flyout>
+        <FlyoutHeader>
+          <FlyoutTitle>{t('devices.details.title')}</FlyoutTitle>
+          <FlyoutCloseBtn onClick={onClose} />
+        </FlyoutHeader>
+        <FlyoutContent>
+          {
+            !device &&
+            <div className="device-details-container">{t("devices.details.noDevice")}</div>
+          }
+          {
+            !!device &&
+            <div className="device-details-container">
+
+              <Grid className="device-details-header">
+                <Row>
+                  <Cell className="col-3"><DeviceIcon type={device.type} /></Cell>
+                  <Cell className="col-7">
+                    <div className="device-name">{device.id}</div>
+                    <div className="device-simulated">{device.isSimulated ? t('devices.details.simulated') : t('devices.details.notSimulated')}</div>
+                    <div className="device-connected">{device.connected ? t('devices.details.connected') : t('devices.details.notConnected')}</div>
+                  </Cell>
+                </Row>
+              </Grid>
+
+              {(!this.state.isAlarmsPending && this.state.alarms && (this.state.alarms.length > 0)) && <RulesGrid {...rulesGridProps} />}
+
+              <FlyoutSection>
+                <SectionHeader>{t('devices.details.telemetry.title')}</SectionHeader>
+                TODO: Add chart when able.
+              </FlyoutSection>
+
+              <FlyoutSection>
+                <SectionHeader>{t('devices.details.tags.title')}</SectionHeader>
+                <SectionDesc>{t('devices.details.tags.description')}</SectionDesc>
+                <Grid className="section-content">
+                  <Row>
+                    <HeaderCell className="col-3">{t('devices.details.tags.keyHeader')}</HeaderCell>
+                    <HeaderCell className="col-7">{t('devices.details.tags.valueHeader')}</HeaderCell>
+                  </Row>
+                  {
+                    (Object.entries(device.tags) || []).map((item, idx) =>
```

Comment:
`.map(([propertyName, propertyValue], idx)`

or something similar, it will be easier to read